### PR TITLE
add flexible-match-range functionality to juniper platform

### PR DESCRIPTION
**Pull request by ryantierney513**
Starting in JUNOS 14, Trio line cards for MX routers now support 'flexible match' which is a very powerful DPI-like capability that allows you to match on a packets' headers or content using byte & bit offsets and hex or binary patterns. This can then be used to combat known attack signatures on a router at near line-rate.
This PR is specific to flexible-match-range; I will probably get working on the other flexible match variant (flexible-match-mask) in January.

### CODE REVIEW

File: lib/policy.py [hunk 1]
```diff
@@ -33,7 +33,16 @@
 
 DEFINITIONS = None
 DEFAULT_DEFINITIONS = './def'
-ACTIONS = set(('accept', 'deny', 'reject', 'next', 'reject-with-tcp-rst'))
+ACTIONS = set(('accept', 'count', 'deny', 'reject', 'next',
+               'reject-with-tcp-rst'))
```
Comment (ryantierney513) [inline lib/policy.py]: This is to resolve a conflict where 'count' was just added via:
https://github.com/google/capirca/commit/74e16b4e378ff96a29d2d09ff91b0858c8bc921e#diff-1b56210f6b46f87bcd833cb0cac39c26

**Comment by sneakywombat** [comment]
Awesome @ryantierney513!

**Comment by ryantierney513** [comment]
github's built in 'Resolve conflicts' feature doesn't ever work (just keeps spinning), so I've manually fixed the conflict for now.

**Comment by ryantierney513** [comment]
hey @ankenyr, have you had a chance to look at this one yet?

File: tests/lib/juniper_test.py [hunk 2]
```diff
@@ -1176,5 +1232,82 @@ def testHopOptProtocol(self):
     output = str(jcl)
     self.failUnless('protocol hop-by-hop;' in output, output)
 
+  def testFlexibleMatch(self):
+    jcl = juniper.Juniper(policy.ParsePolicy(
+                          GOOD_HEADER + GOOD_FLEX_MATCH_TERM,
```
Comment (ankenyr) [inline tests/lib/juniper_test.py]: Please remove 18 spaces from this and the next line.

File: tests/lib/juniper_test.py [hunk 3]
```diff
@@ -1176,5 +1232,82 @@ def testHopOptProtocol(self):
...
+                          self.naming), EXP_INFO)
+    output = str(jcl)
+
+    flexible_match_expected = [
+      'flexible-match-range {',
```
Comment (ankenyr) [inline tests/lib/juniper_test.py]: Add two spaces to line 1243-1248

File: tests/lib/juniper_test.py [hunk 4]
```diff
@@ -1176,5 +1232,82 @@ def testHopOptProtocol(self):
...
+      'match-start payload;',
+      'byte-offset 16;',
+      'bit-offset 7;'
+    ]
+
+    self.assertEquals(all([x in output for x in flexible_match_expected]), True)
+
+  def testFlexibleMatchIPv6(self):
+    jcl = juniper.Juniper(policy.ParsePolicy(
+                          GOOD_HEADER_V6 + GOOD_FLEX_MATCH_TERM,
```
Comment (ankenyr) [inline tests/lib/juniper_test.py]: Same thing here, please remove 18 spaces. You can actually collapse these into a just two lines
    jcl = juniper.Juniper(policy.ParsePolicy(
        GOOD_HEADER_V6 + GOOD_FLEX_MATCH_TERM, self.naming), EXP_INFO)

File: tests/lib/juniper_test.py (hunk 2 above)
Comment (ankenyr) [inline tests/lib/juniper_test.py]: See comment below about making these into two lines

File: tests/lib/juniper_test.py [hunk 5]
```diff
@@ -1176,5 +1232,82 @@ def testHopOptProtocol(self):
...
+                          self.naming), EXP_INFO)
+    output = str(jcl)
+
+    flexible_match_expected = [
+      'flexible-match-range {',
```
Comment (ankenyr) [inline tests/lib/juniper_test.py]: Same thing here, just lint errors, please add two spaces to these lines.

**Comment by ryantierney513** [comment]
Something terrible has happened. Probably going to abort this one and submit a new PR. Not sure where/how a bunch of other commits (and one with a bad email) got picked up.

**Pull request closed without merging**